source/multidimensional-array.d.ts

```diff
@@ -0,0 +1,51 @@
+/**
+Create a type that represents a multidimensional array of the given type and dimension.
+
+Use-cases:
+- Declaring n-dimensional array easily and declaratively.
+- Creating n-dimensional array in functions.
```

Comment:
```suggestion
- Creating a n-dimensional array in functions.
```

Reply:
This item is not much different from the previous one. I would try to list some practical specific use-cases.